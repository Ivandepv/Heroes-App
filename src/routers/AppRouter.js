import React, { useContext } from "react";
import {
  BrowserRouter as Router,
  Switch,
  // Route
} from "react-router-dom";

import { PrivateRoute } from "./PrivateRoute";
import { PublicRoute } from "./PublicRoute";
import { AuthContext } from "../auth/AuthContext";


import { LoginScreen } from "../components/login/LoginScreen";
import { DashboardRoutes } from "./DashboardRoutes";




export const AppRouter = () => {

  const {user:{logged}} = useContext(AuthContext);

    return (
            <Router>
              <div>
                <Switch>
                  <PublicRoute isAuthenticated={logged} exact path="/login" component={LoginScreen} />   

                  <PrivateRoute isAuthenticated={logged} path="/" component={DashboardRoutes} />         
                 </Switch>
              </div>
            </Router>

    )
}
