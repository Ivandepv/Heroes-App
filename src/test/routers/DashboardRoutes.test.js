import { mount } from "enzyme";
import { MemoryRouter } from "react-router-dom";
import { AuthContext } from "../../auth/AuthContext";
import { DashboardRoutes } from "../../routers/DashboardRoutes";


describe('Pruebas en <DashboardRoutes />', ()=>{
    const contextValue = {
        dispatch: jest.fn(),
        user:{
            name: 'Fernando',
            logged: true
        }
        
    }


    test('Debe de mostrarse correctamente', ()=>{

        const wrapper = mount(
        <AuthContext.Provider value= {contextValue}>
            <MemoryRouter>
                <DashboardRoutes />
            </MemoryRouter>
        </AuthContext.Provider>
        );

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe(contextValue.user.name);
    });
})