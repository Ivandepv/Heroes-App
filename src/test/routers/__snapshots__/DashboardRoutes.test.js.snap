// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <DashboardRoutes /> Debe de mostrarse correctamente 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="navbar-collapse"
    >
      <div
        className="navbar-nav"
      >
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/marvel"
          onClick={[Function]}
          style={Object {}}
        >
          Marvel
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/dc"
          onClick={[Function]}
        >
          DC
        </a>
      </div>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </ul>
    </div>
    <div
      className="navbar-collapse collapse w-100 order-3 dual-collapse2"
    >
      <ul
        className="navbar-nav ml-auto"
      >
        <span
          className="nav-item nav-link text-info"
        >
          Fernando
        </span>
        <button
          className="nav-item nav-link btn"
          onClick={[Function]}
        >
          Logout
        </button>
      </ul>
    </div>
  </nav>,
  <div
    className="container mt-2"
  >
    <div>
      <h1>
        Marvel Screen
      </h1>
      <hr />
      <div
        className="card-columns animate__animated animate__fadeIn"
      >
        <a
          href="/hero/marvel-spider"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Spider Man"
              className="img img-responsive"
              src="./assets/heroes/marvel-spider.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Spider Man
                 
              </h5>
              <p
                className="profile-position"
              >
                Peter Parker
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Amazing Fantasy #15
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-captain"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Captain America"
              className="img img-responsive"
              src="./assets/heroes/marvel-captain.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Captain America
                 
              </h5>
              <p
                className="profile-position"
              >
                Steve Rogers
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Captain America Comics #1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-iron"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Iron Man"
              className="img img-responsive"
              src="./assets/heroes/marvel-iron.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Iron Man
                 
              </h5>
              <p
                className="profile-position"
              >
                Tony Stark
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Tales of Suspense #39
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-thor"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Thor"
              className="img img-responsive"
              src="./assets/heroes/marvel-thor.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Thor
                 
              </h5>
              <p
                className="profile-position"
              >
                Thor Odinson
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Journey into Myster #83
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-hulk"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Hulk"
              className="img img-responsive"
              src="./assets/heroes/marvel-hulk.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Hulk
                 
              </h5>
              <p
                className="profile-position"
              >
                Bruce Banner
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        The Incredible Hulk #1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-wolverine"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Wolverine"
              className="img img-responsive"
              src="./assets/heroes/marvel-wolverine.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Wolverine
                 
              </h5>
              <p
                className="profile-position"
              >
                James Howlett
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        The Incredible Hulk #180
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-daredevil"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Daredevil"
              className="img img-responsive"
              src="./assets/heroes/marvel-daredevil.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Daredevil
                 
              </h5>
              <p
                className="profile-position"
              >
                Matthew Michael Murdock
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Daredevil #1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-hawkeye"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Hawkeye"
              className="img img-responsive"
              src="./assets/heroes/marvel-hawkeye.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Hawkeye
                 
              </h5>
              <p
                className="profile-position"
              >
                Clinton Francis Barton
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        Tales of Suspense #57
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-cyclops"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Cyclops"
              className="img img-responsive"
              src="./assets/heroes/marvel-cyclops.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Cyclops
                 
              </h5>
              <p
                className="profile-position"
              >
                Scott Summers
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        X-Men #1
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
        <a
          href="/hero/marvel-silver"
          onClick={[Function]}
        >
          <div
            className="my-card"
          >
            <img
              alt="Silver Surfer"
              className="img img-responsive"
              src="./assets/heroes/marvel-silver.jpg"
            />
            <div
              className=""
            >
              <h5
                className="profile-name"
              >
                 
                Silver Surfer
                 
              </h5>
              <p
                className="profile-position"
              >
                Norrin Radd
              </p>
              <div
                className="profile-overview"
              >
                <div
                  className="profile-overview"
                >
                  <div
                    className="row"
                  >
                    <div
                      className="col-ms-4"
                    >
                      <h3>
                        Marvel Comics
                      </h3>
                      <p>
                        Primera aparición: 
                        <br />
                        The Fantastic Four #48
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>,
]
`;
